test/segment-loader.test.js

```diff
@@ -1145,6 +1145,156 @@ QUnit.module('SegmentLoader', function(hooks) {
       });
     });
 
+    QUnit.test('should use video PTS value for timestamp offset calculation when useDtsForTimestampOffset set as false', function(assert) {
+      loader = new SegmentLoader(LoaderCommonSettings.call(this, {
+        loaderType: 'main',
+        segmentMetadataTrack: this.segmentMetadataTrack,
+        useDtsForTimestampOffset: false
+      }), {});
+
+      const playlist = playlistWithDuration(20, { uri: 'playlist.m3u8' });
+
+      return this.setupMediaSource(loader.mediaSource_, loader.sourceUpdater_).then(() => {
+        return new Promise((resolve, reject) => {
+          loader.one('appended', resolve);
+          loader.one('error', reject);
+
+          loader.playlist(playlist);
+          loader.load();
+
+          this.clock.tick(100);
+
+          standardXHRResponse(this.requests.shift(), videoSegment());
+        });
+      }).then(() => {
+        assert.equal(
+          loader.sourceUpdater_.videoTimestampOffset(),
+          -playlist.segments[0].videoTimingInfo.transmuxedPresentationStart,
```

Comment:
It looks like the numbers for `transmuxedPresentationStart` and `transmuxedDecodeStart` are both the same. If we can mock the values, it might be able to test it better.

Reply:
Agree, I've added a new test asset
here is a commit with changes addressing this core review note:
https://github.com/videojs/http-streaming/pull/1251/commits/337bdc573c13fcd40172a5007e033d6973be6e67